### ISSUE TITLE
Primary navigation of developer.mozilla.org

### ISSUE THREAD

**Issue by scottaohara** [mdn/yari#6870]
### Summary

The primary navigation (references, guides, mdn plus) will display drop down navigations on hover/focus of the primary links.

There are also buttons in small screen view that are presently marked up with aria-haspopup, indicating that they will reveal menu popups.  But the UI that is shown at small screen are better described as disclosure widgets.

At larger screen sizes, these same "menus" do popup on hover/focus of the primary nav link - but these don't behave as menu/menuitems should.  e.g., the link when focused doesn't actually indicate that it has "expanded" to show a sub-navigation.  the next tab stop does bring someone to the sub-nav... again, announced as a menu - but not of the expected keyboard functionalty for a menu exists (arrow keys to navigate).  instead, one must navigate via tab key.  Arrows don't work nor does Esc to close the menu.  There's no way to bypass these menus.  E.g. if i wanted to navigate to "guides" i HAVE to navigate through all of the references sub-navigation items.

Additionally, the menus that popup, while not having the expected keyboard functionality, are also not properly formed menus.

For instance, the following is a reduced version of the rendered markup for the sub-navigation:
```html
<ul id="guides-menu" class="submenu guides hidden inline-submenu-lg" role="menu" aria-labelledby="guides-button">
  <li role="menuitem" class="apis-link-container mobile-only ">
     <a href="/en-US/docs/Learn" class="submenu-item " role="menuitem">
        <div class="submenu-icon learn"></div>
          <div class="submenu-content-container">
            <div class="submenu-item-heading">Overview / MDN Learning Area</div>
            <p class="submenu-item-description">Learn web development</p>
            </div>
       </a>
    </li>
    ....
</ul>
```
Notice that the `<li>` element has a `role=menuitem`, as does it's nested `<a href>` which is incorrect.  The parent menuitem is unnecessary, and can cause for duplicate announcements.  For instance, using Narrator and navigating with Narrator Key + Right arrow, each of these menuitems is found twice due to the duplicate use of `menuitem`.

### URL

https://developer.mozilla.org/en-US/

### Reproduction steps

1. View source of the primary navigation of the mdn website
2. attempt to use it with a screen reader
3. notice that arrow key functionality doesn't work as expected

### Expected behavior

Seems to me that this is just a standard navigation and as such there's no reason to include use of aria-haspopup on the button, nor role=menu and role=menuitems on the sub-navigation list/links.  the removal of those roles would retain the current functionality, but mitigate against exposing roles that indicate different behaviors to how the navigation actually functions.

additionally, per the mention of how the sub-navigation items presently appear on hover/focus, and someone using a keyboard cannot bypass a sub-navigation - see https://www.w3.org/WAI/ARIA/apg/example-index/disclosure/disclosure-navigation-hybrid.html which appears to be the interaction pattern that this navigation should adhere to.

### Actual behavior

See the description in the summary

### Device

Other (specify below)

### Browser

Other (specify below)

### Browser version

Other version (specify below)

### Operating system

Other (specify below)

### Anything else?

per device, browser, os - they're all equally impacted by this issue.

If this is not the correct repo that this bug should have been filed in, please transfer it to the appropriate repro.

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mdn/yari/discussions).

**Comment by caugner** [comment]
Hi @scottaohara, thank you for taking the time to report this accessibilty issue. 👏 

If I interpret your analysis correctly, there are two measures that we should consider taking:
1. Remove aria-haspopup from the button, role=menu from the ul, and role=menuitems from the submenu li/a.
2. Add support for arrow navigation between (left/right) and within (up/down) those menus, and for closing the currently opened menu with Escape.

As for 1., we would implement that asap, as it's a low-hanging fruit and improves the UX for folks with screen readers.
As for 2., we would probably accept a community PR as long as it doesn't increase complexity significantly.

Would it make sense to add the roles back once 2. is implemented?
And is there anything I missed to address?

**Comment by scottaohara** [comment]
Hi @caugner 
re: 1 - yes. 
re: 2 - would be a bonus for keyboard UX, and in lieu of adding a button to purposefully show/hide the sub-navigations on larger screens, yes this would be needed.  but alternatively, arrow key navigation could be considered a usability improvement if a button to explicitly open/close the sub-navigations were instead added to the top nav, so that someone navigating with tab key did not have to navigate through sub-navigations to reach the next top-level navigation item.
E.g., if I want to get to "guides" i have to press tab 7 times to navigate through the sub-navigation for references.  If I want to get to MDN Plus - even though it's the third top level nav item, that's 13 presses of the tab key to get there.

**Comment by caugner** [comment]
I have now fixed 1. via https://github.com/mdn/yari/pull/6929, which inadvertently closed this issue, but actually I will leave it closed, as adding arrow/Escape navigation to the menu is non-trivial and not on our agenda.

That being said, we would accept a community PR as long as it doesn't increase complexity significantly.
